presto-spi/src/main/java/com/facebook/presto/spi/function/QualifiedFunctionName.java

```diff
@@ -33,11 +33,13 @@ private QualifiedFunctionName(CatalogSchemaName functionNamespace, String functi
         this.functionName = requireNonNull(functionName, "name is null").toLowerCase(ENGLISH);
     }
 
+    // Do not use this function, it is only used for test or json deserialization
     @JsonCreator
     public static QualifiedFunctionName of(String dottedName)
     {
         String[] parts = dottedName.split("\\.");
         if (parts.length != 3) {
+            // this should never happen
```

Comment:
I think `IlligalArgumentException` means that this should never happen. So this is not necessary.